### PR TITLE
Refactor knative test: abridge test by removing unnecessary assertion.

### PR DESCRIPTION
**Pull request by pratikjagrut**
### Motivation

Refactor knative test

### Testing
`go test ./pkg/controller/servicebindingrequest -run TestKnativeServicesContractWithBinder -v` 
or
`make test-unit` to run all the unit tests

### CODE REVIEW

**Comment by sbose78** [comment]
Curious, why is it unnecessary?

**Comment by pratikjagrut** [comment]
> Curious, why is it unnecessary?

@sbose78 a few days back I asked @isutton about why tests in `binder_test` do not check binder.Bind(). So he said Bind() is tested in reconciler_test, here we just need to check if the resources are being available or not. So to make this knative test similar to all other tests I replaced binding assertion with the searching assertion.

File: pkg/controller/servicebindingrequest/binder_test.go [hunk 1]
```diff
@@ -300,19 +300,11 @@ func TestKnativeServicesContractWithBinder(t *testing.T) {
...
-
-	c := corev1.Container{}
-	u := containers[0].(map[string]interface{})
-	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u, &c)
-	require.NoError(t, err)
-	require.Equal(t, name, c.EnvFrom[0].SecretRef.Name)
+	t.Run("Knative service contract with service binding operator", func(t *testing.T) {
+		list, err := binder.search()
+		assert.Nil(t, err)
+		assert.Equal(t, 1, len(list.Items))
```
Comment (akashshinde) [inline pkg/controller/servicebindingrequest/binder_test.go]: @pratikjagrut should we consider adding more assertion or is checking item length is enough ?

Comment (pratikjagrut) [reply pkg/controller/servicebindingrequest/binder_test.go]: Yeah, kinda it's enough. search will add found knative service in the list, and we are creating only one service at that time, so the length is gonna be 1, and if it is greater, then something is fishy, which is not part of this test, and can be seen as a bug or test failure. SO I feel this assertion is enough and although all other present unit tests are tested on same assertion.

Comment (pratikjagrut) [reply pkg/controller/servicebindingrequest/binder_test.go]: The aim here is just to make sure resources are being available.

**Comment by pradeepto** [comment]
@sbose78 @baijum @akashshinde @otaviof Can we get a closure on this?

File: pkg/controller/servicebindingrequest/binder_test.go (hunk 1 above)
Comment (isutton) [reply pkg/controller/servicebindingrequest/binder_test.go]: I'm fine with this change. An issue should be created to include a case where `matchLabels` is `nil` and multiple knative services.

**Pull request merged**